### PR TITLE
What's New for ECSV format in io.ascii

### PR DESCRIPTION
**Pull request by taldcroft**
Note that this contains a slightly unrelated fix to the ECSV examples which I noticed along the way.  I decided to save some travis time this way, esp. since the doc examples in question are skipped.

### CODE REVIEW

File: docs/whatsnew/1.0.rst [hunk 1]
```diff
@@ -146,6 +149,55 @@ fast version, in which case the pure-Python version will automatically be used.
...
+meta-data such as comments, keywords and column data types, units, and
+descriptions.  Using the newly defined `Enhanced Character Separated Values
+format <https://github.com/astropy/astropy-APEs/blob/master/APE6.rst>`_ it is
+now possible to write a table to an ASCII-format file and read it back with no
+loss of information.  The ECSV format has been designed to be both
+human-readable and compatible with most simple CSV readers.
+
+In the example below we show writing a table that has ``float32`` and ``bool``
+types.  This illustrates the simple look of the format which has a few header
+lines (starting with ``#``) in `YAML <http://www.yaml.org/>`_ format and then
```
Comment (mdboom) [inline docs/whatsnew/1.0.rst]: Do we want to mention that the new feature requires PyYAML?  I know it's mentioned in `install.rst`, but maybe here as well to help people who just want to start playing with this?

Comment (taldcroft) [inline docs/whatsnew/1.0.rst]: Yes, good point.

**Comment by astrofrog** [comment]
@taldcroft - just a reminder to implement @mdboom's comment in the next day or so, so we can merge and backport it to the 1.0.x branch before the final release.

**Comment by taldcroft** [comment]
@astrofrog - done.

**Comment by astrofrog** [comment]
Thanks!

**Comment by astrofrog** [comment]
I've backported this in the v1.0.x branch as fe161268379faed95fd521323ba509384381c52f

**Pull request merged**